consensus/core/src/linearizer.rs

```diff
@@ -68,6 +75,11 @@ impl Linearizer {
                             !committed.contains(ancestor)
                                 && last_committed_rounds[ancestor.author] < ancestor.round
                         })
+                        .filter(|ancestor| {
```

Comment:
Since we are doing this change, can we migrate away from skipping blocks when `reached a round that we already committed` as well? Changing this will be necessary for fastpath too. We can combine the filter expressions and have something like:
```
if committed.contains(ancestor) {
  return false;
}
if gc_enabled {
  ...
} else {
  last_committed_rounds[ancestor.author] < ancestor.round
}
```

Reply:
I might be missing something , but why is this desired? It looks like that way it's possible to recommit already committed blocks. For fast path IIUC the worry might be about disconnected that cary votes but eventually don't make it into that DAG due to this rule?

Reply:
Recommit is impossible because of the `if committed.contains(ancestor)` check. The goal of the `last_committed_rounds[ancestor.author] < ancestor.round` check is to avoid committed equivocating blocks aggressively. But with fastpath it can drop "finalized" blocks similar to setting GC depth less than voting rounds.

Reply:
The `committed` struct is a temporary local one that refers to the currently committed sub dag only. When we attempt to commit the next sub dag that resets and we won't be able to tell whether a block has already been committed or not.

Reply:
Then we will have to keep track of the commit status for blocks above the GC round. We can change this separately but it will be worth leaving a TODO.

Reply:
> Then we will have to keep track of the commit status for blocks above the GC round

yep that would be necessary

> We can change this separately but it will be worth leaving a TODO.

Ok I'll leave a TODO for now and come back to that on follow up PR as some additional work will be needed. But good call. I see the FP case.